crates/eips/src/eip4844/sidecar.rs

```diff
@@ -54,6 +55,24 @@ impl core::fmt::Debug for BlobTransactionSidecar {
     }
 }
 
+impl BlobTransactionSidecar {
+    /// Matches versioned hashes and constructs `BlobAndProofV1` objects.
+    pub fn match_versioned_hashes(&self, versioned_hashes: &[B256]) -> Vec<Option<BlobAndProofV1>> {
+        let mut result = vec![None; versioned_hashes.len()];
```

Comment:
this alloc is a bit unfortunate because this isn't very cheap because we need to alloc `len(versioned_hashes)` for each `BlobTransactionSidecar`

I think what we can do here instead is make this return an iterator of `(j, BlobAndProofV1)` because we're only interested in the index of the matching hash in the versioned_hashes slice